src/components/Explore/Map.tsx

```diff
@@ -135,16 +135,22 @@ const ExploreMap = (): JSX.Element => {
                 if (e.features && e.features[0]) {
                     const feature = e.features[0];
                     const stationProperties = feature.properties as StationSummary;
-                    const newSelectedStation =
+                    let newSelectedStation =
                         stationProperties.name === selectedStationRef.current?.name ? null : stationProperties;
+                    if (newSelectedStation) {
+                        const index: number = stationsList
+                            .map((station) => station.name)
+                            .indexOf(newSelectedStation.name);
+                        newSelectedStation = stationsList[index];
+                    }
                     dataActionDispatcher({
                         type: 'updateSelectedStation',
                         station: newSelectedStation
                     });
                     selectedStationRef.current = newSelectedStation;
                     if (map.getZoom() < 6 && newSelectedStation) {
```

Comment:
Now that we have the `flyTo` part in the hook on `selectedStation` change, we can remove this block. Otherwise, the map calls `flyTo` twice when users click on a station.